### PR TITLE
Add books to author

### LINKED ISSUE #110: An Author should have a set of books
**Describe the solution you'd like**
A `Book` currently has an `Author`, but an `Author` does not contain any books.

The `Author` entity should have a set of books (one-to-many).

A `Set` should be used instead of a `List` as it would be better to find duplicates than have an ordering. 

There should also be a JPA query for retrieving all of the books by particular author (see `PredefinedShelfRepository` for a similar query but on shelves)

### PR DESCRIPTION
**Pull request by jmprathab**
Author has a set of books.
Add a repository query to fetch all books by an author

Closes #110

## Summary of change

Add set of books to Authors

## Related issue

knjk04/book-project#110

### CODE REVIEW

**Comment by knjk04** [comment]
Thanks for this. I'll review this pull request later this week

**Comment by knjk04** [comment]
It's been a while since have heard from you. How are you getting on with this issue? 

If you need help, let us know over Gitter. Alternatively, if you need more time, just let us know. 

(This is just a friendly checkup, as per our contributing document).

**Comment by knjk04** [comment]
Please also note that this is now blocking #114 , which is a high-priority bug

**Comment by jmprathab** [comment]
@knjk04 Sorry, I missed your change request notification.
Can you suggest the changes? Do you want me to construct `Author.java` with an empty Set instead of passing around null?

**Comment by knjk04** [comment]
No matter.

It would be better if the author constructor remained untouched -- i.e. it only takes a first & last name as arguments.

File: src/main/java/com/karankumar/bookproject/backend/entity/Author.java [hunk 1]
```diff
@@ -32,7 +38,7 @@
 @Data
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
 @AllArgsConstructor
-@EqualsAndHashCode(callSuper = true)
+@EqualsAndHashCode(callSuper = true, exclude= "books")
 public class Author extends BaseEntity {
```
Comment (jmprathab) [inline src/main/java/com/karankumar/bookproject/backend/entity/Author.java:42]: @knjk04 To solve the requirement, 

I will have to include a hand-written constructor which takes only firstName and lastName as parameters. (books is going to be initialized with `new HashSet<>()`)

Please let me know if the approach is fine. I am not aware of any way to skip a particular field from `@AllArgsConstructor`

Comment (knjk04) [reply src/main/java/com/karankumar/bookproject/backend/entity/Author.java:42]: That's fine.

Also, could you please resolve the merge confilcts?

Thanks!

File: src/main/java/com/karankumar/bookproject/backend/entity/Author.java [hunk 2]
```diff
@@ -31,17 +38,25 @@
 @Entity
 @Data
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
-@AllArgsConstructor
-@EqualsAndHashCode(callSuper = true)
+@EqualsAndHashCode(callSuper = true, exclude= "books")
 public class Author extends BaseEntity {
 
+    public Author(String firstName, String lastName){
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/backend/entity/Author.java:44]: there should be a space after the closing parenthesis, as per our style guide

**Comment by knjk04** [comment]
Thanks for this. There was one style guide violation (a missing space), which I have now resolved.

**Pull request merged**